docs/tutorial.md

```diff
@@ -1,333 +1,373 @@
+* `name` is an identifier of *any solid*, but in this case it is an identifier of `polyline`.
+It's type is `String`. **This value can be required by any solid;
+you can set it, you can not to set it, but without you won't be able to control solid, since it won't be inherited.**
+
+This is an example of polyline with other solid `box`:
+```kotlin
+box(100, 100, 100, name = "box"){
+   x = -10
+   y = -10
+   z = -10
+   opacity = 0.4
+}
+polyline(Point3D(30, 20, 10), Point3D(30, -100, 30), Point3D(30, -100, 30), Point3D(50, -100, 30), name = "polyline"){
+   color("red")
+}
+```
+
+![](../docs/images/polyline-points.png)
+![](../docs/images/polyline-points-2.png)
+
+### 2) Box
+
+First thing which has to be mentioned is that `box` takes four values: `box(x, y, z, name)`
+* `x` - x-axis length of the `box`
+* `y` - y-axis length of the `box`
+* `z` - z-axis length of the `box`
+
+These values have `Float` type. 
+
+*`x`, `y`, and `z` are necessary values, which cannot be ignored. You have to set them.*
+
+* `name` - `box`'es identifier. You've already met it. 
+
+Let's create just usual `box` with equal ribs.
+
+```kotlin
+   box(50, 50, 50, name = "box") {
+        color("pink")
+   }
+```
+   ![](../docs/images/box.png)
+
+Now, let's make `box` with bigger `y` value.
+   ```kotlin
+   box(10, 25, 10, name = "high box") {
+        color("black")
+   }
+   ```
+As you can see, only rib of `y-axis` differs from other ribs.
+
+   ![](../docs/images/high-box.png)
+
+For final trial, let's create `box` with bigger `x` value.
+   
+   ```kotlin
+   box(65, 40, 40, name = "wide box") {
+        x = 0
+        y = 0
+        z = 0
+        color("black")
+   }
+   ```
+Predictably, only `x-axis` rib bigger than other ribs. 
+
+   ![](../docs/images/wide-box.png)
+   
+### 3) Sphere
+
+It takes in two values: `radius`, and `name`. 
+We bring you to mind that `name` is a general value for all solids, so do not wonder, since all solids need their own identifier.
+
+As for `radius`, it has `Float` type, and, as you can guess, it sets radius of the sphere, which will be created.
+   ```kotlin
+   sphere(50, name = "sphere") {
+        x = 0
+        y = 0
+        z = 0
+        opacity = 0.9
+        color("blue")
+   }
+   ```
+   ![](../docs/images/sphere.png)
+
+### 4) Hexagon
+
+It is solid which has six edges. It is set by eight values: `node1`,..., `node8`. They all have `Point3D` type, so they are just points, vertices.
+
+*Six edges are these:* 
+1) Edge with vertices `node1`, `node4`, `node3`, `node2`
+2) Edge with vertices `node1`, `node2`, `node6`, `node5`
+3) Edge with vertices `node2`, `node3`, `node7`, `node6`
+4) Edge with vertices `node4`, `node8`, `node7`, `node3`
+5) Edge with vertices `node1`, `node5`, `node8`, `node4`
+6) Edge with vertices `node8`, `node5`, `node6`, `node7`
+
+![](../docs/images/scheme.png)
+
+As hexagon takes in specific points, we understand that this solid cannot be moved, it fixed in space, and it can't make pivots.
```

Comment:
```suggestion
As the hexagon takes in specific points, we understand that this solid cannot be moved, it is fixed in space, and it can't make pivots.
```